cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/SourcePointClientTest.java

```diff
@@ -1,262 +1,250 @@
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = Build.VERSION_CODES.O_MR1)
+public class SourcePointClientTest {
+
+    private OkHttpClient http = new OkHttpClient();
+    private SourcePointClient sourcePointClientMock;
+    private GDPRConsentLib.OnLoadComplete onLoadComplete;
+
+    private MockWebServer webServerMock;
+
+    private String consentUUID = "consentUUID";
+    private String euConsent = "euConsent";
+    private String meta = "meta";
+    private boolean isNative = false;
+    private NetworkInfo networkInfo;
+    private ConnectivityManager connectivityManager;
+
+
+    @Before
+    public void setUp() throws IOException {
+        onLoadComplete = mock(GDPRConsentLib.OnLoadComplete.class);
+        int accountId = 123;
+        int propertyId = 1234;
+        String property = "example.com";
+        String targettingParams = "targetingParams";
+        String authId = "";
+        boolean staging = false;
+        boolean stagingCampaign = false;
+        PropertyConfig propertyConfig = new PropertyConfig(accountId,propertyId,property,"123221");
+        SourcePointClientConfig config = new SourcePointClientConfig(propertyConfig,stagingCampaign,staging,targettingParams,authId);
+
+        networkInfo = mock(NetworkInfo.class);
+        when(networkInfo.isConnectedOrConnecting()).thenReturn(true);
+        when(networkInfo.getType()).thenReturn(ConnectivityManager.TYPE_MOBILE);
+        when(networkInfo.isRoaming()).thenReturn(true);
+        connectivityManager = mock(ConnectivityManager.class);
+        when(connectivityManager.getActiveNetworkInfo()).thenReturn(networkInfo);
+        sourcePointClientMock =  new SourcePointClient(http,config, connectivityManager);
+        webServerMock = new MockWebServer();
+        webServerMock.start();
+    }
+
+    @After
+    public void finish() throws IOException {
+        webServerMock.shutdown();
+        reset(onLoadComplete);
+        reset(networkInfo);
+        reset(connectivityManager);
+    }
+
+    @Test
+    public void noInternetErrorWhileGetMessge() {
+        when(networkInfo.isConnectedOrConnecting()).thenReturn(false);
+        webServerMock.enqueue(new MockResponse().setBody("{}").setResponseCode(200));
+        sourcePointClientMock.baseMsgUrl = webServerMock.url("/").toString();
+
+        try {
+            sourcePointClientMock.getMessage(isNative, consentUUID, meta, euConsent, onLoadComplete);
+            ArgumentCaptor<Object> captor = ArgumentCaptor.forClass(Object.class);
+            verify(onLoadComplete, timeout(1000)).onSuccess(captor.capture());
```

Comment:
why do we need this timeout here?

Reply:
Okhttp is performing operation in background so we need to little timeout to wait for okhttp thread to respond.